src/SFML/Graphics/Font.cpp

```diff
@@ -129,13 +129,6 @@ m_pixelBuffer(copy.m_pixelBuffer)
 }
 
 
-////////////////////////////////////////////////////////////
-Font::~Font()
-{
-    cleanup();
```

Comment:
We don't need to call this anymore...?

Reply:
Also I think you need to keep this in the source file to avoid breaking the Android build.

Reply:
https://github.com/SFML/SFML/blob/709530d062ac9c452f50b06d37cdd800c2f9ba68/src/SFML/Graphics/Font.cpp#L522-L531

`Font::cleanup` doesn't do anything that the destructors of class members aren't already doing.

> Also I think you need to keep this in the source file to avoid breaking the Android build.

Can you help me understand why this is? Why isn't the implicitly defined destructor good enough?

Reply:
> Can you help me understand why this is? Why isn't the implicitly defined destructor good enough? Would it have something to do with the to move-related special member functions that are defaulted but can now be removed?

If you have

```cpp
// someclass.h

struct SomeDependency;

struct SomeClass
{
    std::unique_ptr<SomeDependency> m_dep;
};
```

that will fail to compile because `std::unique_ptr<SomeDependency>::~unique_ptr()` is going to attempt to call `SomeDependency::~SomeDependency()` which is not available in scope.

If you instead do:

```cpp
// someclass.h

struct SomeDependency;

struct SomeClass
{
    std::unique_ptr<SomeDependency> m_dep;
    ~SomeClass();
};


// someclass.cpp
#include <someclass.h>
#include <somedependency.h>

SomeClass::~SomeClass() = default;
```

then the destructor of `std::unique_ptr` will have visibility of `SomeDependency::~SomeDependency()`.

Reply:
So the destructor is necessary because we choose to use forward declarations instead of including the header?

Reply:
> So the destructor is necessary because we choose to use forward declarations instead of including the header?

Yes, and we chose the latter to improve compilation speed and reduce physical dependencies between SFML's components.